crates/providers/blobstore-s3/src/lib.rs

```diff
@@ -0,0 +1,468 @@
+
+//         let ld = self.get_ld(&ctx).await?;
+//         let receiver = InvocationHandler::new(&ld);
+
+//         let container_id = chunk.container_id.clone();
+//         let object_id = chunk.object_id.clone();
+//         let actor_id = &ld.actor_id;
+//         let chunk_len_bytes: u64 = chunk
+//             .bytes
+//             .len()
+//             .try_into()
+//             .map_err(|e| ProviderInvocationError::Provider(format!("failed to do: {e}")))?;
+
+//         let cr = receiver.receive_chunk(chunk).await
+//             .map_err(|e| ProviderInvocationError::Provider(format!(
+//                 "sending chunk error: Container({container_id}) Object({object_id}) to Actor({actor_id}): {e:?}",
+//                 )))?;
+
+//         Ok(if cr.cancel_download {
+//             0
+//         } else {
+//             chunk_len_bytes
+//         })
+//     }
+// }
+
+// #[async_trait]
+// impl WasmcloudCapabilityProvider for BlobstoreS3Provider {
+//     /// The fs provider has one configuration parameter, the root of the file system
+//     async fn put_link(&self, ld: &LinkDefinition) -> bool {
+//         for val in ld.values.iter() {
+//             info!("ld conf {:?}", val);
+//         }
+
+//         // Determine the root path value
+//         let root_val: PathBuf = match ld.values.iter().find(|(key, _)| key == "ROOT") {
+//             None => "/tmp".into(),
+//             Some((_, value)) => value.into(),
+//         };
+
+//         // Build configuration for FS Provider to use later
+//         let config = BlobstoreS3ProviderConfig {
+//             ld: ld.clone(),
+//             root: root_val.clean(),
+//         };
+
+//         info!("Saved BlobstoreS3ProviderConfig: {:#?}", config);
+//         info!(
+//             "File System Blob Store Container Root: '{:?}'",
+//             &config.root
+//         );
+
+//         // Save the configuration for the actor
+//         self.config
+//             .write()
+//             .await
+//             .insert(ld.actor_id.clone(), config.clone());
+
+//         // Resolve the subpath from the root to the actor ID, carefully
+//         let actor_dir = match self.resolve_subpath(&config.root, &ld.actor_id).await {
+//             Ok(path) => path,
+//             Err(e) => {
+//                 error!("Failed to resolve subpath to actor directory: {e}");
+//                 return false;
+//             }
+//         };
+
+//         // Create directory for the individual actor
+//         match create_dir_all(actor_dir.as_path()).await {
+//             Ok(()) => true,
+//             Err(e) => {
+//                 error!("Could not create actor directory: {:?}", e);
+//                 false
+//             }
+//         }
+//     }
+
+//     async fn delete_link(&self, actor_id: &str) {
+//         self.config.write().await.remove(actor_id);
+//     }
+
+//     async fn shutdown(&self) {
+//         self.config.write().await.drain();
+//     }
+// }
```

Comment:
If this code is unused, we can just remove it

Reply:
Yup, this code got ported -- removed this!